### PR TITLE
General fixes for the 0.10 release, including:

### PR DESCRIPTION
**Pull request by joehuchette**
* More Correct sparse matmul code
* Fixes for some vectorized operators
* Avoiding map for explicit loops in operators
* Small tweaks to parseExpr

### CODE REVIEW

**Comment by mlubin** [comment]
Is this passing locally on 0.3? I'm having trouble because of https://github.com/JuliaLang/FactCheck.jl/issues/62, but travis doesn't seem to care?

**Comment by joehuchette** [comment]
I haven't ran this against v0.3 yet

**Comment by joehuchette** [comment]
Passing locally on v0.3, but I also haven't done a ``Pkg.update`` for a while.

**Comment by mlubin** [comment]
Never mind, was a local issue.

**Comment by mlubin** [comment]
Some of the ``_multiply!`` methods don't seem to be covered: https://codecov.io/github/JuliaOpt/JuMP.jl/src/operators.jl?ref=6f2df681b7ee3231f53ee821f2f6cdee1c318a89

File: src/v0.4/parseExpr_staged.jl [hunk 1]
```diff
@@ -529,13 +524,13 @@ function parseExpr(x, aff::Symbol, lcoeffs::Vector, rcoeffs::Vector, newaff::Sym
             s = gensym()
             newaff_, parsed = parseExprToplevel(x.args[2], s)
             push!(blk.args, :($s = 0.0; $parsed))
-            push!(blk.args, :($newaff = $(Expr(:call,:*,lcoeffs...,newaff_,newaff_,rcoeffs...)) + $aff))
+            push!(blk.args, :($newaff = $aff + $(Expr(:call,:*,lcoeffs...,newaff_,newaff_,rcoeffs...))))
```
Comment (mlubin) [inline src/v0.4/parseExpr_staged.jl]: What difference does this make?

Comment (joehuchette) [inline src/v0.4/parseExpr_staged.jl]: More consistent with my mental model that the update is ``aff + c*x``?

Comment (mlubin) [inline src/v0.4/parseExpr_staged.jl]: Addition is always commutative, but more consistency is okay

Comment (joehuchette) [inline src/v0.4/parseExpr_staged.jl]: Yeah, it's more for the generated code

**Comment by mlubin** [comment]
CI failures on 0.3.

**Comment by joehuchette** [comment]
Dispatch is different, apparently
```
.*(A::SparseMatrixCSC{Tv,Ti<:Integer},B::Array{T,N})
```
is more specific than
```
(.*){S<:Number,T<:JuMPTypes,N}(lhs::AbstractArray{S,N},rhs::AbstractArray{T,N})
```

**Comment by mlubin** [comment]
Any easy workaround?

**Comment by joehuchette** [comment]
Add more methods like ``.*(A::SparseMatrixCSC{Tv,Ti<:Integer},B::Array{T,N})``? I'm inclined to leave it as-is and see if anyone complains before adding more code, though.

**Comment by mlubin** [comment]
So run the offending tests on 0.4 only?

**Comment by joehuchette** [comment]
Yeah

**Comment by mlubin** [comment]
Also, the fuzzer throws an error instead of using FactCheck, which kills travis: https://travis-ci.org/JuliaOpt/JuMP.jl/jobs/78066233#L6727

**Comment by IainNZ** [comment]
Is today the day?

**Comment by joehuchette** [comment]
OK, I think 0.10 is g2g. Who wants to do the honors?

**Comment by mlubin** [comment]
image

**Comment by mlubin** [comment]
@joehuchette, you can do the honors. This was mostly your work.

**Comment by joehuchette** [comment]
I'll tag if you write the mailing list email :)

**Comment by mlubin** [comment]
Deal

**Comment by IainNZ** [comment]
I should update that OR-Exchange post I guess

**Comment by IainNZ** [comment]
And the JuliaOpt.org table

**Pull request merged**